A6.md

```diff
@@ -0,0 +1,457 @@
+
+The only RPCs that are counted as failures for the throttling policy are RPCs that fail with a status code that qualifies as a [retryable](#retryable-status-codes) or [non-fatal status code](#hedging-policy), or that receive a pushback response indicating not to retry. This avoids conflating server failure with responses to malformed requests (such as the `INVALID_ARGUMENT` status code).
+
+#### Pushback
+Servers may explicitly pushback by setting metadata in their response to the client. The pushback can either tell the client to retry after a given delay or to not retry at all. If the client has already exhausted its `maxRetryAttempts`, the call will not be retried even if the server says to retry after a given delay.
+
+Pushback may also be received to a hedged request. If the pushback says not to retry, no further hedged requests will be sent. If the pushback says to retry after a given delay, the next hedged request (if any) will be issued after the given delay has elapsed.
+
+A new metadata key, `"x-grpc-retry-pushback-ms"`, will be added to support server pushback. If the value for pushback is set to -1, then it will be seen as the server asking the client not to retry at all.
+
+#### Summary of Retry and Hedging Logic
+There are five possible types of server responses. The list below enumerates the behavior of retry and hedging policies for each type of response. In all cases, if the maximum number of retry attempts or the maximum number of hedged requests is reached, no further RPCs are sent. Hedged RPCs are returned to the client application when all outstanding and pending requests have either received a response or been canceled.
+
+1. OK
+  1. Retry policy: Successful response, return success to client application
+  2. Hedging policy: Successful response, cancel previous and pending hedges
+2. Retryable/Non-Fatal Status Code
+  1. Retry policy: Retry according to policy
+  2. Hedging policy: Immediately send next scheduled hedged request, if any. Subsequent hedged requests will resume at `hedgingDelayMs`
+3. Fatal Status Code
+  1. Retry policy: Don't retry, return failure to client application
+  2. Hedging policy: Cancel previous and pending hedges
+4. Pushback: Don't retry
+  1. Retry policy: Don't retry, return failure to client application
+  2. Hedging policy: Don’t send any more hedged requests.
+5. Pushback: Retry in *n* ms
+  1. Retry policy: Retry in *n* ms. If this attempt also fails, retry delay will reset to initial backoff for the following retry (if applicable)
+  2. Hedging policy: Send next hedged request in *n* ms. Subsequent hedged requests will resume at `n + hedgingDelayMs`
+
+![State Diagram](A6_graphics/StateDiagram.png)
+
+[Link to diagram on draw.io](https://drive.google.com/a/google.com/file/d/0Bwcp6fL6ikUvOGJWcExrNFhMZ0k/view?usp=sharing)
```

Comment:
I do think it would be better to include the SVG files in this PR instead of linking to an external source.  Otherwise, there's no guarantee that someone won't delete the external source without realizing that it's referenced from this document.